velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -466,6 +470,32 @@ struct TimestampMinusIntervalDayTime {
   }
 };
 
+template <typename T>
+struct TimestampAtTimezoneFunction : public TimestampWithTimezoneSupport<T> {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<TimestampWithTimezone>& result,
+      const arg_type<Timestamp>& ts,
+      const arg_type<Varchar>& timeZone) {
+
+    auto timeZoneStr = std::string_view(timeZone.data(), timeZone.size());
+    auto zone = date::locate_zone(timeZoneStr);
```

Comment:
there is an conversion operator between StringView and std::string_view. Likely you can just:

`auto zone = date::locate_zone(timeZone);`

Reply:
I did this manual conversion because it was unable to automatically cast the input Varchar/`velox::StringView` to `std::string_view`.